docs/quality-assurance-test-cases.md

```diff
@@ -173,7 +173,29 @@ This document contains some examples of test cases for each feature implemented
 | Logged in user tries to complete a task from an non existing request (The request  exists in a different relationship) | Fail |
 | Logged in user tries to complete a task from a request which is not in the ACCEPTED state (as a mentor or as a mentee) | Fail |
 
+### List All Tasks
+**Service:** PUT /mentorship_relation/{request_id}/task/{task_id}
 
+| Test Case  | Outcome |
+| ---------- | -----------|
+| Logged in user requests list all task,or out of all task, those which involves the user (as a mentor or as a mentee) | Success |
+| Logged in user requests list all task,or out of all task, those which involves the user (neither as a mentor nor as a mentee) | Fail |
+| Logged in user requests to list all task whose id does not exist, which involves the user (as a mentor or as a mentee) | Fail |     
+| Logged in user requests list all task,or out of all task, those which doesn't involves the user (as a mentor or as a mentee) | Fail |
+| Logged in user requests to list all task which is invalid (not an integer), which involves the user(as a mentor or as a mentee) | Fail |
+| Not logged in user (invalid token) tries to List all task| Fail |
+
+### Tasks
```

Comment:
This should come even before List all tasks.
`Task` will contain `Create` and `List All Tasks` as subheadings.